### PR TITLE
chore: adding 'no-experimental-fetch' node option by default

### PR DESCRIPTION
**Pull request by rusackas**
### SUMMARY
When running `npm run build` I would get this error:
```
throw new Error("You must provide the URL of lib/mappings.wasm by calling " +
            ^

Error: You must provide the URL of lib/mappings.wasm by calling SourceMapConsumer.initialize({ 'lib/mappings.wasm': ... }) before using SourceMapConsumer
    at readWasm (/Users/evan/GitHub/superset/superset-frontend/node_modules/react-hot-loader/node_modules/source-map/lib/read-wasm.js:8:13)
    at wasm (/Users/evan/GitHub/superset/superset-frontend/node_modules/react-hot-loader/node_modules/source-map/lib/wasm.js:25:16)
    at /Users/evan/GitHub/superset/superset-frontend/node_modules/react-hot-loader/node_modules/source-map/lib/source-map-consumer.js:264:14

Node.js v18.16.1
```

When troubleshooting, I found a helpful note in the docs:

> If this type of error comes while building assets(i.e using above commands):
> 
> ```bash
> Error: You must provide the URL of lib/mappings.wasm by calling SourceMapConsumer.initialize
> ```
> 
> Then put this:
> 
> ```bash
> export NODE_OPTIONS=--no-experimental-fetch
> ```

~~I'm _assuming_ that nobody _needs_ the `experimental-fetch` option enabled, so rather than letting users hit this error and go looking for the fix, let's just add the fix to the webpack build scripts in `package.json`~~

Updated approach! Upon fiddling with things, it seems that simply _installing_ the `source-map` dependency seems to automagically provide the missing `mappings.wasm` file, and we no longer have a problem.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

n/a

### TESTING INSTRUCTIONS

run all three build methods and make sure they don't bomb out :)

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by rusackas** [comment]
@mdeshmu since you added the docs (thank you!!!) let me know if I'm missing something here... I assume this would just fix it, but let us know if you think this PR sets up some other danger/error.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -598,18 +598,6 @@ There are three types of assets you can build:
 2. `npm run dev-server`: local development assets, with sourcemaps and hot refresh support
 3. `npm run build-instrumented`: instrumented application code for collecting code coverage from Cypress tests
 
-If this type of error comes while building assets(i.e using above commands):
-
-```bash
-Error: You must provide the URL of lib/mappings.wasm by calling SourceMapConsumer.initialize
```
Comment (mdeshmu) [inline CONTRIBUTING.md:604]: Even though i didn't add these docs but I feel `calling SourceMapConsumer.initialize` would still be needed.

Comment (rusackas) [reply CONTRIBUTING.md:604]: Upon messing with it, it seems like you shouldn't _have to_ call the initialize method when using React... and that merely installing the `source-map` npm module fixes this. 

I did `unset NODE_OPTIONS` to set my machine back to "stock" and all of the webpack build modes seem to work fine now.

File: superset-frontend/package.json [hunk 2]
```diff
@@ -208,6 +208,7 @@
     "rison": "^0.1.1",
     "scroll-into-view-if-needed": "^2.2.28",
     "shortid": "^2.2.6",
+    "source-map": "^0.7.4",
```
Comment (michael-s-molina) [inline superset-frontend/package.json:211]: Shouldn't this be a dev dependency?

Comment (rusackas) [reply superset-frontend/package.json:211]: I feel like I tried that and it didn't work for some reason. The sourcemaps are generated/used by prod builds created by `npm run build` so maybe it's used in rendering/utilizing the source maps in those builds? I'm not exactly sure, but I somewhat recall it throwing the same error when it was a dev dependency.

Comment (rusackas) [reply superset-frontend/package.json]: Ok, it's now a dev dependency :D Seems to work!

Comment (rusackas) [reply superset-frontend/package.json]: You were right... n`pm run dev`, `npm run dev-server`, and `npm run buid` all seem to work fine for me with it as a dev dependency. Game on!

**Review by michael-s-molina — APPROVED** [review]
LGTM

**Pull request merged**